File: LeetCode/Problems/0020_Valid_Parentheses/valid_parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        dict = {
            "}" : "{",
            "]" : "[",
            ")" : "("
        }
        pStack = []
        for c in s:
            if c in dict.values():
                pStack.append(c)
            else:
                if not pStack or dict[c] != pStack.pop():
                    return False
        return not pStack;

File: LeetCode/Problems/0020_Valid_Parentheses/valid_parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        std::stack<char> pStack;
        for(const auto& c : s){
            switch(c){
                case '{':
                    pStack.push('}');
                    break;
                case '[':
                    pStack.push(']');
                    break;
                case '(':
                    pStack.push(')');
                    break;
                default:
                    if(pStack.size() == 0 || c != pStack.top())
                        return false;
                    else
                        pStack.pop();
            }
        }
        return (pStack.size() == 0) ? true : false;
    }
};
